source/agora/test/EnrollmentManager.d

```diff
@@ -96,3 +96,50 @@ unittest
                  node.getPreimage(enroll_3.utxo_key) != PreImageInfo.init,
             5.seconds));
 }
+
+// Test for restoring enrollments and pre-images from the Genesis block
+unittest
+{
+    import agora.consensus.validation.PreImage;
+    import core.thread;
+    import std.conv;
+
+    const validator_cycle = 20;
+    auto params = new immutable(ConsensusParams)(validator_cycle);
+    auto network = makeTestNetwork(TestConf.init, params);
+    scope(exit) network.shutdown();
+    scope(failure) network.printLogs();
+    network.start();
+    network.waitForDiscovery();
+
+    // Boilerplate
+    auto nodes = network.clients;
+    auto node_1 = nodes[0];
+
+    // Check if genesis block has enrollments
+    const b0 = node_1.getBlocksFrom(0, 2)[0];
+    assert(b0.header.enrollments.length == 4);
+    auto enroll = b0.header.enrollments[0];
+
+    // Check if the `node_1` restores block from the chain after restarting
+    network.restart(node_1);
+    network.waitForDiscovery();
+    containSameBlocks(nodes, 0).retryFor(5.seconds);
```

Comment:
That's not really checking anything. We haven't created any block, so there is nothing to restore.